pootle/apps/pootle_store/models.py

```diff
@@ -865,17 +865,21 @@ def toggle_qualitycheck(self, check_id, false_positive, user):
             sub_type = SubmissionTypes.MUTE_CHECK
         else:
             sub_type = SubmissionTypes.UNMUTE_CHECK
-
-        sub = Submission(creation_time=make_aware(timezone.now()),
-                         translation_project=self.store.translation_project,
-                         submitter=user, field=SubmissionFields.NONE,
-                         unit=self, store=self.store, type=sub_type,
-                         quality_check=check)
+        update_time = make_aware(timezone.now())
+        sub = Submission(
+            creation_time=update_time,
+            translation_project=self.store.translation_project,
+            submitter=user,
+            field=SubmissionFields.NONE,
+            unit=self,
+            store=self.store,
+            type=sub_type,
+            quality_check=check)
         sub.save()
 
         # update timestamp
         # log user action
-        self.save()
+        self.save(submitted_by=user, submitted_on=update_time)
```

Comment:
@phlax it'd be great to see when and why we set submitted_by described in one place.
previously you could get a real translator via `submitted_by`. I'm not sure if it's expected that you set this field when muting/unmuting checks.